src/app/Plugins/Navigation/Components/navigation/navigation.component.ts

```diff
@@ -61,8 +60,8 @@ export class NavigationComponent implements OnInit {
     this.showMode = false;
   }
 
-  async bestType(from: string, to: string, date: string, hour: number, minute: number) {
-    this.createRoute(from, to, date, hour, minute);
+  async bestType(from: string, to: string, date: string) {
+    this.createRoute(from, to, date);
     this.navigationService.best(this.navPathData).subscribe((data: BestTransportResult) => this.bestTransport = { ...data });
     await new Promise((resolve, reject) => setTimeout(resolve, 3000));
```

Comment:
use the callback don't use setTimeout

Reply:
But without waiting a few seconds there will be an error on first use/click.

Reply:
i know you have to wait for the response of the request, but therefore use the callback of the subscribe method, it is called when the request is finished